src/tactic/solve_by_elim.lean

```diff
@@ -39,42 +57,75 @@ do (hs, gex, hex, all_hyps) ← decode_simp_arg_list hs,
    else return hs
 
 /--
-The internal implementation of `solve_by_elim`, with a limiting counter.
+Configuration options for `solve_by_elim`.
+
+* `accept : list expr → tactic unit` determines whether the current branch should be explored.
+   At each step, before the lemmas are applied,
+   `accept` is passed the original metavariables (as a `list expr` argument)
+   reported by `get_goals` when `solve_by_elim` started,
+   (these metavariables may by now have been partially solved by previous `apply` steps).
```

Comment:
```suggestion
   `accept` is passed the proof terms for the original goals
   reported by `get_goals` when `solve_by_elim` started.
   These proof terms may be metavariables (if no progress has been made on that goal)
   or may contain metavariables at some leaf nodes 
   (if the goal has been partially solved by previous `apply` steps).
```
?

Reply:
I'll make one further change to your suggestion.